File: test-plugins/org.yakindu.sct.generator.java.test/test-gen/org/yakindu/sct/generator/java/test/StaticChoiceTest.java
/**
 * Copyright (c) 2015 committers of YAKINDU and others.
 * All rights reserved. This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License v1.0
 * which accompanies this distribution, and is available at
 * http://www.eclipse.org/legal/epl-v10.html
 *
 * Contributors:
 *     committers of YAKINDU - initial API and implementation
 */

package org.yakindu.sct.generator.java.test;

import org.junit.After;
import org.junit.Before;
import org.junit.Test;
import static org.junit.Assert.*;
import org.yakindu.scr.staticchoice.StaticChoiceStatemachine;
import org.yakindu.scr.staticchoice.StaticChoiceStatemachine.State;
/**
 *  Unit TestCase for StaticChoice
 */
@SuppressWarnings("all")
public class StaticChoiceTest {

	private StaticChoiceStatemachine statemachine;

	@Before
	public void setUp() {
		statemachine = new StaticChoiceStatemachine();
		statemachine.init();
	}

	@After
	public void tearDown() {
		statemachine = null;
	}

	@Test
	public void testStaticChoiceTest() {
		statemachine.enter();
		assertTrue(statemachine.isStateActive(State.main_region_Start));
		statemachine.runCycle();
	}
}

File: test-plugins/org.yakindu.sct.generator.cpp.test/gtests/StaticChoiceTest/StaticChoiceTest.cc
/**
* Copyright (c) 2017 committers of YAKINDU and others.
* All rights reserved. This program and the accompanying materials
* are made available under the terms of the Eclipse Public License v1.0
* which accompanies this distribution, and is available at
* http://www.eclipse.org/legal/epl-v10.html
*
* Contributors:
*     committers of YAKINDU - initial API and implementation
*/
#include <string>
#include "gtest/gtest.h"
#include "StaticChoice.h"
#include "sc_types.h"
StaticChoice* statechart;

class StatemachineTest : public ::testing::Test{
	protected:
	virtual void SetUp() {
		statechart = new StaticChoice();
		statechart->init();
	}
	virtual void TearDown() {
		delete statechart;
	}
};


TEST_F(StatemachineTest, StaticChoiceTest) {
	
	
	statechart->enter();
	
	EXPECT_TRUE(statechart->isStateActive(StaticChoice::main_region_Start));
	
	statechart->runCycle();
	
}
